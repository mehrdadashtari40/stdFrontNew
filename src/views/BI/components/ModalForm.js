import React, {Component} from 'react';
import LanguageStore from "../../../components/i18n/LanguageStore";
import Form from "./Form";
import { withTranslation } from 'react-i18next';

export default class ModalForm extends Component {

	onSubmit = () => {
		this.props.submit();
	}

	render() {
		let formElements = null;
		if (this.props.edit) {
		    formElements = <Form edit={true} structure={this.props.structure} />;
		} else {
		    formElements = <Form edit={false} structure={this.props.structure} />;
		}

		return(<form id="new-user-form" className="smart-form" noValidate="novalidate">
				<div>
					<fieldset>
						<div className="row" style={{marginTop:'10px'}}>
							{formElements}
						</div>
					</fieldset>
					<footer>
						<button type="button" onClick={ () => this.onSubmit() } className="btn btn-success"><i
							className="fa fa-check"></i>
							{this.props.t('Submit')}
						</button>
					</footer>
				</div>
			</form>)
    }
}


