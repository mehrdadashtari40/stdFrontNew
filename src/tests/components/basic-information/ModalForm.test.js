import React from 'react';
import { shallow } from 'enzyme';

import ModalForm from '../../../views/basic-information/components/ModalForm';

jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));

test('Make Snapshot file , DaynamicDataTable Component', () => {
	const wrapper = shallow(<ModalForm />);

	expect(wrapper).toMatchSnapshot();
});
