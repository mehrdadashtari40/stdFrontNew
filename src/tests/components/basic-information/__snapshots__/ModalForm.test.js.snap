// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , DaynamicDataTable Component 1`] = `
<form
  className="smart-form"
  id="new-user-form"
  noValidate="novalidate"
>
  <div>
    <fieldset>
      <div
        className="row"
        style={
          Object {
            "marginTop": "10px",
          }
        }
      >
        <Form
          edit={false}
        />
      </div>
    </fieldset>
    <footer>
      <button
        className="btn btn-success"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fa fa-check"
        />
      </button>
    </footer>
  </div>
</form>
`;
